lewis_emulators/cryogenic_sms/interfaces/stream_interface.py

```diff
@@ -0,0 +1,248 @@
+                                            .build(),
+            CmdBuilder(self.write_output_mode).spaces().regex("T(?:ESLA)*").spaces().arg("OFF|ON|0|1").spaces().eos()
+                                              .build(),
+            CmdBuilder(self.write_ramp_target).spaces().regex("R(?:AMP)*").spaces().arg("ZERO|MID|MAX|0|%|!")
+                                              .spaces().eos().build(),
+            CmdBuilder(self.write_heater_status).spaces().regex("H(?:EATER)*").spaces().arg("OFF|ON|1|0").spaces()
+                                                .eos().build(),
+            CmdBuilder(self.write_pause).spaces().regex("P(?:AUSE)*").spaces().arg("OFF|ON|0|1").spaces().eos().build(),
+            CmdBuilder(self.write_heater_value).regex(re_set).escape("H(?:EATER)*").spaces().float().spaces().eos()
+                                               .build(),
+            CmdBuilder(self.write_max_target).regex(re_set).choice("MAX", "!").spaces().float().spaces().eos().build(),
+            CmdBuilder(self.write_mid_target).regex(re_set).choice("MID", "%").spaces().float().spaces().eos().build(),
+            CmdBuilder(self.write_ramp_rate).regex(re_set).regex("R(?:AMP)*").spaces().float().spaces().eos().build(),
+            CmdBuilder(self.write_limit).regex(re_set).regex("L(?:IMIT)*").spaces().float().spaces().eos().build(),
+            CmdBuilder(self.write_constant).regex(re_set).regex("T(?:PA)*").spaces().float().spaces().eos().build()
+        }
+
+    def _timestamp(self):
+        return datetime.now().strftime('%H:%M:%S')
+
+    def _create_log_message(self, pv, value, suffix=""):
+        current_time = self._timestamp()
+        self._device.log_message = "{} {}: {}{}\r\n".format(current_time, pv, value, suffix)
+
+    def handle_error(self, request, error):
+        self.log.error("Error occurred at {}: {}".format(request, error))
+        print("Error occurred at {}: {}".format(request, error))
+
+    def _get_output_mode_string(self):
+        return "TESLA" if self._device.is_output_mode_tesla else "AMPS"
+
+    def _get_paused_state_str(self):
+        return "ON" if self._device.is_paused else "OFF"
+
+    def _get_ramp_target_value(self):
+        target = None
+        if self._device.ramp_target == "MID":
+            target = self._device.mid_target
+        elif self._device.ramp_target == "MAX":
+            target = self._device.max_target
+        return target
+
+    def read_direction(self):
+        dir = self._device.direction
+        if dir == "+":
+            dir_string = "POSITIVE"
+        elif dir == "-":
+            dir_string = "NEGATIVE"
+        elif dir == "0":
+            dir_string = "ZERO"
+        return "CURRENT DIRECTION: {}\r\n".format(dir_string)
+
+    def write_direction(self, direction):
+        self._device.direction = direction
+        return "........\r\n"

```

Comment:
It seems that a lot of return statements are of the form `.....{}\r\n` can we refactor this return format into a variable? Also, if all commands end with `\r\n` this should be the out terminator